### ISSUE TITLE
implementing arduino client interface

### ISSUE THREAD

**Issue by narangmayank** [espressif/arduino-esp32#7917]
### Board

ESP32

### Device Description

Development Board

### Hardware Configuration

None

### Version

latest development Release Candidate (RC-X)

### IDE Name

Arduino IDE

### Operating System

Windows

### Flash frequency

40 MHz

### PSRAM enabled

yes

### Upload speed

115200

### Description

can we make the arduino-esp32 core based on arduino client interface so that the core functionality can work on any standard arduino client interface.

for eg,
 
In HTTPUpdate Library we can do something like this,

t_httpUpdate_return update(Client& client, const String& url, const String& currentVersion = "");

Basically, I was doing HTTP OTA via GSM and then I get to know that it works only on WiFiClient interface, that does make sense because ESP32 inside is a WiFi Soc but what if we can make it out to generic so that any network interface will work.

any thoughts?

### Sketch

```cpp
nothing here
```

### Debug Message

```plain
nothing here
```

### Other Steps to Reproduce

nothing here

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by me-no-dev** [comment]
This would be doable, if `Client` has all necessary methods. It is a good proposal and worth looking into.

**Comment by mayankbytebeam** [comment]
I’m glad you liked it.

Seems like you added this to Arduino ESP32 Core Project Roadmap, but I couldn't find it.

**Comment by me-no-dev** [comment]
In the project roadmap you need to select `3.0.0 Release` and it would be at the bottom of the `ToDo` column

**Comment by VojtechBartoska** [comment]
Postponing this to 3.1.0 Milestone.

**Comment by Parsaabasi** [comment]
Hello, 

Due to the overwhelming volume of issues currently being addressed, we have decided to close the previously received tickets. If you still require assistance or if the issue persists, please don't hesitate to reopen the ticket.

Thanks.
